notebooks/utils/phenopacket.py

```diff
@@ -0,0 +1,208 @@
+import json
+import logging
+import os
+from typing import Any, List
+
+from google.protobuf.json_format import ParseDict, ParseError
+from phenopackets import Phenopacket as ProtoPhenopacket
+
+logger = logging.getLogger(__name__)
+
+
+class InvalidPhenopacketError(ValueError):
+    """
+    Exception raised when a JSON payload does not conform to minimal
+    Phenopacket expectations.
+    """
+
+
+class Phenopacket:
+    """
+    Helper for loading and querying GA4GH Phenopacket JSON files.
+
+    Wraps the Protobuf-based Phenopacket model from the `phenopackets` package,
+    providing minimal on-disk loading, basic schema validation via Protobuf,
+    and simple phenotype queries.
+    """
+
+    def __init__(self, phenopacket_json: dict[str, Any]) -> None:
+        """
+        Initialize Phenopacket from a JSON-decoded dictionary.
+
+        Performs minimal structural validation (presence and type of
+        `phenotypicFeatures`) before storing the raw dictionary for
+        downstream queries.
+
+        Parameters
+        ----------
+        phenopacket_json : dict
+            A JSON-decoded dict representing a GA4GH Phenopacket. It must
+            contain a key `"phenotypicFeatures"` mapped to a list of feature
+            objects, each of which should be a dict with at least:
+
+            {
+                "type": {
+                    "id": "<HPO:NNNNNNN>",
+                    "label": "<Phenotype Label>"
+                }
+            }
+
+        Raises
+        ------
+        InvalidPhenopacketError
+            If `phenopacket_json` is not a dict or if
+            `"phenotypicFeatures"` is missing or not a list.
+        """
+        self._validate_structure(phenopacket_json)
```

Comment:
You introduced two seperate validation mechanism. This one and this:

ParseDict(data, ProtoPhenopacket())

Think we only need the later one.